src/api/spec/models/workflow_run_spec.rb

```diff
@@ -86,4 +88,24 @@
       end
     end
   end
+
+  describe '#scm_vendor' do
+    let(:workflow_run) { create(:workflow_run, request_headers: headers) }
+
+    context 'when event is from GitHub' do
+      let(:headers) { "HTTP_X_GITHUB_EVENT_TYPE: pull_request\nHTTP_X_GITHUB_EVENT: pull_request\nHTTP_ACCEPT: application/xml" }
+
+      it 'identifies GitHub' do
+        expect(workflow_run.scm_vendor).to be(:github)
```

Comment:
```suggestion
      subject { workflow_run.scm_vendor }
      it 'identifies GitHub' do
        expect(subject).to be(:github)
```
Then you can reuse the subject on the other context too. Using the subject block helps identifying what do you want to test.